extensions/amp-accordion/0.1/amp-accordion.js

```diff
@@ -50,19 +66,63 @@ class AmpAccordion extends AMP.BaseElement {
       header.setAttribute('role', 'tab');
       content.classList.add('-amp-accordion-content');
       content.setAttribute('role', 'tabpanel');
-      content.setAttribute(
-          'aria-expanded', section.hasAttribute('expanded').toString());
       let contentId = content.getAttribute('id');
       if (!contentId) {
         contentId = this.element.id + '_AMP_content_' + index;
         content.setAttribute('id', contentId);
       }
+      if (this.currentState_[contentId]) {
+        section.setAttribute('expanded', '');
+      } else if (this.currentState_[contentId] == false) {
+        section.removeAttribute('expanded');
+      }
+      content.setAttribute(
+          'aria-expanded', section.hasAttribute('expanded').toString());
       header.setAttribute('aria-controls', contentId);
       header.addEventListener('click', boundOnHeaderClick_);
     });
   }
 
   /**
+   * Generate a sessionStorage Key based on amp-accordion element id.
+   * @return {string}
+   * @private
+   */
+  getSessionStorageKey_() {
+    const id_ = this.element.id;
+    const url = removeFragment(this.win_.location.href);
+    return `amp-${id_}-${url}`;
+  }
+
+  /**
+   * Get previous state from sessionStorage.
+   * @return {!Object|null}
+   * @private
+   */
+  getSessionState_() {
+    try {
+      const sessionStr = this.win_./*OK*/sessionStorage.getItem(this.id_);
```

Comment:
Please file an issue to centralize the implementation.

On Fri, Jun 24, 2016 at 5:55 PM, erwin mombay <notifications@github.com>
wrote:

> In extensions/amp-accordion/0.1/amp-accordion.js
> <https://github.com/ampproject/amphtml/pull/3300#discussion_r68480733>:
>
> > +   * @private
> > +   */
> > +  getSessionStorageKey_() {
> > +    const id_ = this.element.id;
> > +    const url = removeFragment(this.win_.location.href);
> > +    return `amp-${id_}-${url}`;
> > +  }
> > +
> > +  /**
> > +   * Get previous state from sessionStorage.
> > +   * @return {!Object|null}
> > +   * @private
> > +   */
> > +  getSessionState_() {
> > +    try {
> > +      const sessionStr = this.win_./*OK*/sessionStorage.getItem(this.id_);
>
> right, the reason why is that this is now cropping up on production with
> new errors and would prefer that it at least have the same failure
> mechanism (in storage-impl.js case it throws then is caught and becomes a
> noop)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/3300/files/ab86aaa43118807a096a2ea24c9253e56b362059#r68480733>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAFeT13A8Ej5_MntYLqL5aPsfJoQQzISks5qPHxygaJpZM4IlCv0>
> .
>